nncf/quantization/algorithms/weight_compression/torch_fx_backend.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import torch
+import torch.fx
+
+import nncf
+import nncf.errors
+import nncf.tensor
+from nncf.common.graph.definitions import NNCFGraphNodeType
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.operator_metatypes import CONST_NOOP_METATYPES
+from nncf.common.graph.operator_metatypes import OperatorMetatype
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.experimental.common.tensor_statistics.collectors import TensorCollector
+from nncf.experimental.torch.fx import operator_metatypes as fx_om
+from nncf.experimental.torch.fx.commands import FXApplyTransformationCommand
+from nncf.experimental.torch.fx.model_transformer import FXModelTransformer
+from nncf.experimental.torch.fx.node_utils import get_graph_node_by_name
+from nncf.experimental.torch.fx.node_utils import get_tensor_constant_from_node
+from nncf.experimental.torch.fx.transformations import constant_update_transformation_builder
+from nncf.experimental.torch.fx.transformations import module_insertion_transformation_builder
+from nncf.parameters import CompressWeightsMode
+from nncf.quantization.algorithms.weight_compression.backend import WeightCompressionAlgoBackend
+from nncf.quantization.algorithms.weight_compression.config import WeightCompressionParameters
+from nncf.quantization.algorithms.weight_compression.lora_correction import LoraCorrectionAlgorithm
+from nncf.quantization.algorithms.weight_compression.weight_lowering import compress_weight
+from nncf.tensor import Tensor
+from nncf.tensor.definitions import TensorDataType
+from nncf.torch.graph import operator_metatypes as om
+from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.model_graph_manager import find_const_node_in_constant_subgraph
+from nncf.torch.model_graph_manager import get_const_node
+from nncf.torch.quantization.layers import AsymmetricWeightsDecompressor
+from nncf.torch.quantization.layers import SymmetricWeightsDecompressor
+from nncf.torch.tensor_statistics.collectors import get_raw_stat_collector
+
+
+class FXWeightCompressionAlgoBackend(WeightCompressionAlgoBackend):
```

Comment:
@anzr299 please suggest how to avoid copy/past code from PTWightCompressionAlgoBackend.

Reply:
For the constant assignments?

Reply:
As I understand constant assignment has different implementations in PyTorch and PyTorch FX backends. 

This is general comment because I see that a lot of code from PyTorch backend  are used without changes in the Torch FX backend.

Reply:
Yes, for some functions there are minimal changes. I will try to minimize and reuse the code from PyTorch backend.

Reply:
Can we inherit `PTWeightCompressionAlgoBackend` here, is it a good idea?
CC: @alexsu52